File: java/classes/inheritance/MultilevelInheritance.java
class Mammal {
  public boolean isDomesticated;
  public boolean hasTail;

  public static boolean isMilkFeeded = true;
  public static String classTaxon = "Mammalia";
}

class Fox extends Mammal {
  public String name;

  public static boolean hasFur = true;
  public static String speciesTaxon = "Vulpes vulpes";

  public Fox() {
    hasTail = true;
    isDomesticated = false;
  }

  public void show() {
    System.out.println("Hi, my name is " + name
      + ".\nClass: " + classTaxon
      + "\nSpecies: " + speciesTaxon
      + "\nAm I milk-feeded as a cub? " + isMilkFeeded
      + "\nDo I have tail? " + hasTail
      + "\nDo I have fur? " + hasFur
      + "\nAm I domesticated? " + isDomesticated
    );
  }
}

class PetFox extends Fox {
  public PetFox() {
    isDomesticated = true;
  }
}

class MultilevelInheritance {
  public static void main(String[] args) {
    Fox foxFerdinand = new Fox();
    foxFerdinand.name = "Ferdinand";

    foxFerdinand.show();
    System.out.println();

    PetFox foxAgnes = new PetFox();
    foxAgnes.name = "Agnes";

    foxAgnes.show();
    System.out.println();
  }
}

File: c#/classes/inheritance/MultilevelInheritance.cs
class Mammal {
  public bool IsDomesticated;
  public bool HasTail;

  public static bool IsMilkFeeded = true;
  public static string ClassTaxon = "Mammalia";
}

class Fox : Mammal {
  public string Name;

  public static bool HasFur = true;
  public static string SpeciesTaxon = "Vulpes vulpes";

  public Fox() {
    HasTail = true;
    IsDomesticated = false;
  }

  public void Show() {
    System.Console.WriteLine("Hi, my name is " + Name
      + ".\nClass: " + ClassTaxon
      + "\nSpecies: " + SpeciesTaxon
      + "\nAm I milk-feeded as a cub? " + IsMilkFeeded
      + "\nDo I have tail? " + HasTail
      + "\nDo I have fur? " + HasFur
      + "\nAm I domesticated? " + IsDomesticated
    );
  }
}

class PetFox : Fox {
  public PetFox() {
    IsDomesticated = true;
  }
}

class MultilevelInheritance {
  public static void Main(string[] args) {
    Fox foxFerdinand = new Fox();
    foxFerdinand.Name = "Ferdinand";

    foxFerdinand.Show();
    System.Console.WriteLine();

    PetFox foxAgnes = new PetFox();
    foxAgnes.Name = "Agnes";

    foxAgnes.Show();
    System.Console.WriteLine();
  }
}
